modules/andes-core/broker/src/main/java/org/wso2/andes/store/cassandra/CQLBasedMessageStoreImpl.java

```diff
@@ -654,29 +660,84 @@ public AndesMessageMetadata getMetaData(long messageId) throws AndesException {
 
+        if (firstMsgId == 0) {
+            firstMsgId = ServerStartupRecoveryUtils.getStartMessageIdForWarmStartup();
+        }
+        long messageIdDifference = ServerStartupRecoveryUtils.getMessageDifferenceForWarmStartup();
+        lastMsgId = firstMsgId + messageIdDifference;
+
         try {
             Statement statement = QueryBuilder.select().column(CQLConstants.METADATA).column(CQLConstants.MESSAGE_ID).
                     from(config.getKeyspace(), CQLConstants.METADATA_TABLE).
                     where(eq(CQLConstants.QUEUE_NAME, storageQueueName)).
                     and(gte(CQLConstants.MESSAGE_ID, firstMsgId)).
+                    and(lte(CQLConstants.MESSAGE_ID, lastMsgId)).
                     limit(count).
                     setConsistencyLevel(config.getReadConsistencyLevel());
 
             ResultSet resultSet = execute(statement, "retrieving metadata list from " + storageQueueName +
-                    " with starting msg id " + firstMsgId + " and limit " + count);
-            List<AndesMessageMetadata> metadataList =
+                    " with starting msg id " + firstMsgId + " and ending message id " + lastMsgId);
+            metadataList =
                     new ArrayList<AndesMessageMetadata>(resultSet.getAvailableWithoutFetching());
 
             for (Row row : resultSet) {
                 metadataList.add(getMetadataFromRow(row, row.getLong(CQLConstants.MESSAGE_ID)));
             }
-            return metadataList;
+
+            if (metadataList.size() == 0) {
+                readingCassandraInfoLog(timer);
+                while (lastMsgId <= lastRecoveryMessageId) {
+                    long nextMsgId = lastMsgId + 1;
+                    lastMsgId = nextMsgId + messageIdDifference;
+                    statement = QueryBuilder.select().column(CQLConstants.METADATA).column(CQLConstants.MESSAGE_ID).
+                            from(config.getKeyspace(), CQLConstants.METADATA_TABLE).
+                            where(eq(CQLConstants.QUEUE_NAME, storageQueueName)).
+                            and(gte(CQLConstants.MESSAGE_ID, nextMsgId)).
+                            and(lte(CQLConstants.MESSAGE_ID, lastMsgId)).
+                            limit(count).
+                            setConsistencyLevel(config.getReadConsistencyLevel());
+
+                    resultSet = execute(statement, "retrieving metadata list from " + storageQueueName +
+                            " with starting msg id " + firstMsgId + " ending message id " + lastMsgId);
+                    metadataList =
+                            new ArrayList<AndesMessageMetadata>(resultSet.getAvailableWithoutFetching());
+
+                    for (Row row : resultSet) {
+                        metadataList.add(getMetadataFromRow(row, row.getLong(CQLConstants.MESSAGE_ID)));
+                    }
+                    if (metadataList.size() > 0) {
+                        break;
```

Comment:
Same explanation given in above HectorBasedMessageStoreImpl.java is apply to here as well.